pallets/liquidity-pools-gateway/queue/src/benchmarking.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_traits::liquidity_pools::test_util::Message as LPTestMessage;
+use frame_benchmarking::{account, impl_benchmark_test_suite, v2::*};
+use frame_system::RawOrigin;
+use parity_scale_codec::EncodeLike;
+
+use super::*;
+
+#[benchmarks(
+	where
+		T: Config<Message = LPTestMessage>,
```

Comment:
This will probably not pass for the real runtimes. My suggestion is to add a `Default` bound for Message only when benchmarking it. (which would be an `Invalid` one)

Option2. Add a worst-case method under `runtime-benchmarks` in `LPEncoding`, which allows creation of the worst message possible.

Option3. Add a parameter to the benchmarks that allows to pass the `Message`, so each message is benchmarked for itself. This will be the most accurate/clean, but not sure if this is possible to pass non-integer parameters to benchmarking functions 🤔

Reply:
I would go with option 2 since that makes the most sense for benchmarks IMO.

Reply:
> so each message is benchmarked for itself.
Would this be possible? More specifically, does the benchmark test suite execute for each variant of an enum?

Reply:
The benchmark doesn't know the `Message` itself. But if it receive the Message as a parameter it would compute the effect of that `Message` for that specific call

Reply:
But not sure if it would work. Option 2 seems good

Reply:
Oh, but wait. I forgot about - https://github.com/centrifuge/centrifuge-chain/blob/66e3252e13e3c632142c0af22d2acfc3e68515ab/pallets/liquidity-pools-gateway/queue/src/mock.rs#L59

Isn't it easier if we just return some actual weight here that is then used by the benchmarks?

Reply:
That's other option. Just the weight you think is higher than that call. We do that in liquidity-pools related pallets

Reply:
Done, added a default weight ref time that's also used in the pallet instead of default. PTAL

Reply:
Ok, just one thing, the issue is still there because the benchmark is tied to `LPTestMessage`

Reply:
Been trying things out for a while now and I'm still a bit confused...

Option 2 doesn't really work since `LPEncoding` is not a trait used as a constraint for the `Message` type of this pallet, as per [these constraints](https://github.com/centrifuge/centrifuge-chain/blob/8e5171185b78f3b1cfb894f0b756430e43790303/pallets/liquidity-pools-gateway/queue/src/lib.rs#L53).

> Ok, just one thing, the issue is still there because the benchmark is tied to LPTestMessage

This will remain true if we use the mocked runtime anyway. The message type that we use [here](https://github.com/centrifuge/centrifuge-chain/blob/8e5171185b78f3b1cfb894f0b756430e43790303/pallets/liquidity-pools-gateway/queue/src/mock.rs#L57) is the `LPTestMessage`.

Reply:
Updated this btw, took me a while to figure out what was happening when running tests and then the check benchmark scripts.